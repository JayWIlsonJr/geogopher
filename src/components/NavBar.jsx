import React from 'react';
import { NavLink } from 'react-router-dom';

export default class NavBar extends React.Component {

  render() {
    return (
      <div>

        <nav>
            <NavLink exact to="/" > games list </NavLink>
            <NavLink to="/map"> map </NavLink>
            <NavLink to="/login"> map </NavLink>
        </nav>

      </div>

    )
  }
}
