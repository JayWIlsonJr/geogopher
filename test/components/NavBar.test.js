/*
 * NavBar.test.js
 */

import React from 'react';
import ReactDOM from 'react-dom';
import NavBar from '../../src/components/NavBar';
import renderer from 'react-test-renderer';
import { Link } from 'react-router-dom';
import { StaticRouter } from 'react-router';
import { createRenderer } from 'react-test-renderer/shallow';

global.requestAnimationFrme = function(callback) {
  setTimeout(callback, 0);
};

describe('the map', () => {

  /* These first two tests are essentially the same. The test using
   * createRenderer does a shallow rendering, which means that children of
   * the component will not be rendered.  Use TestUtils instead.
   */

  it('renders NavBar component', () => {
    const div = document.createElement('div');
    ReactDOM.render(<navbar />, div);
  });

  it('renders a NavBar component', () => {
    const navBarRenderer = createRenderer();
    navBarRenderer.render(<NavBar/>);
    let n = navBarRenderer.getRenderOutput();
    expect(n.type).toBe('div');
  });

  it('home link matches snapshot', () => {
    const context = {}
    const tree = require('react-test-renderer').create(
      <StaticRouter location="http://localhost:1337/" context={context}>
        <Link to="/" />
      </StaticRouter>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('map link matches snapshot', () => {
    const context = {}
    const tree = require('react-test-renderer').create(
      <StaticRouter location="http://localhost:1337/map" context={context}>
        <Link to="/map" />
      </StaticRouter>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

});
