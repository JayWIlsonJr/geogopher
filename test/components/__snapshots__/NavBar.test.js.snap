// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the map home link matches snapshot 1`] = `
<a
  href="/"
  onClick={[Function]}
/>
`;

exports[`the map map link matches snapshot 1`] = `
<a
  href="/map"
  onClick={[Function]}
/>
`;
